fix(hostRules): sort by matchHost length (#14264)
renovatebot/renovate: docs/usage/configuration-options.md

}
```

> If multiple `hostRules` match a request, then they will be applied in the following order/priority, where latest takes precedence:

Reviewer: "Where latest takes precedence" is hard to understand for me, we have a list, that I assume is in the priority order? But then our text says "where latest takes precedence" which seems to imply that the last item on the list has the biggest priority?

> If multiple `hostRules` match a request, then they will be applied in the following order/priority, where latest takes precedence:

Author: ```suggestion
If multiple `hostRules` match a request, then they will be applied in the following order/priority:
```

- rules with only `hostType` specified

> - rules with only `matchHost` specified (sorted by length of `matchHost` string)

Author: ```suggestion
- rules with only `matchHost` specified (sorted by `matchHost` length if multiple match)
```

> - rules with both `matchHost` and `hostType` specified (sorted by length of `matchHost` string)

Reviewer: I think it would be easier to understand this list if it was numbered in order of priority:

1. First rule that matches
2. Second rule that matches
3. Third rule that matches

> - rules with both `matchHost` and `hostType` specified (sorted by length of `matchHost` string)

Author: ```suggestion
- rules with both `matchHost` and `hostType` specified (sorted by `matchHost` length if multiple match)
```
